### PR TITLE
feature/owoify

### PR DESCRIPTION
**Pull request by not-archiey**
This pull request adds the command `owoify` which turns a normal text into an "owo"-ed text

Before : 
Are we still friends? I...I-I understand if you never want to see me again

After `owoify` : 
Are we stiww fwiends? I...I-I understand if u nyewer want two see mwe again

### CODE REVIEW

File: commands/Owoify.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      category: "fun",
+      description: "this cwommand will owoify ywour text",
+      usage: `Run \`owoify <text>\` or \`owoify <intensity> <text>\` to owoify any text!\nIntensities available are \`easy\` (default), \`medium\`, and \`hard\`.`
+    });
+  }
+
+  run(client, message, args) {
+    let level = "owo";
+
+    if (intensities[args[0]]) {
```
[the remark this answers is not in the document]
Comment (Leguu) [reply commands/Owoify.js:23]: If it doesn't exist, then the message is empty, ask the user to provide an input text.

File: commands/Owoify.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      level = intensities[args[0]];
+      args.shift();
```
Comment (Leguu) [inline commands/Owoify.js:25]: `Array#shift()` returns the element anyway, so replace `args[0]` here with it.

File: commands/Owoify.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  medium: "uwu",
+  hard: "uvu"
+};
+
+export default class Owoify extends Command {
+  constructor() {
+    super({
+      name: "owoify",
+      category: "fun",
+      description: "this cwommand will owoify ywour text",
```
Comment (Leguu) [inline commands/Owoify.js:15]: Don't owoify the description or the usage text.

**Comment by not-archiey** [comment]
Still need to add emotes check

File: commands/Owoify.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    let cursedText = [];
+
+    args.forEach(arg => {
+      if (!EMOTE_ID_REGEX.test(arg)) {
+        cursedText.push(owoify.default(arg, level).replace(/`/g, "\\`"));
+      } else {
+        cursedText.push(arg);
+      }
+    });
```
Comment (duncte123) [inline commands/Owoify.js:42]: This is just bad performance-wise, you should just replace the brackets that it replaces with a global replace at the and

File: commands/Owoify.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import owoify from "owoify-js";
+import Command from "./base/Command.js";
+import { EMOTE_ID_REGEX } from "../utils/discordPatterns.js";
+
+const intensities = {
+  easy: "owo",
+  medium: "uwu",
+  hard: "uvu"
+};
```
Comment (duncte123) [inline commands/Owoify.js:9]: Make this a private property of the class

File: commands/Owoify.js (hunk 4 above)
Comment (not-archiey) [reply commands/Owoify.js:42]: I can't write a more effective algorithm than this, can you help me? 😅

Comment (duncte123) [reply commands/Owoify.js:42]: `.replace(/(whatever brackets are converted to/g, "<")` and for the other one as well

File: commands/Owoify.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        `${message.author}, can you please provide me a message? <:YoichiPlease:692008252690530334>`
+      );
+    }
+
+    if (this.intensities[args[0]]) {
+      level = this.intensities[args.shift()];
+    }
+
+    let cursedText = "";
+    let charCount = 0;
```
Comment (duncte123) [inline commands/Owoify.js:35]:
```suggestion
    const charCount = args.map(a => a.lenght).reduce((a, b) => a + b, 0);
```

File: commands/Owoify.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+    args.forEach(arg => {
+      if (!EMOTE_ID_REGEX.test(arg)) {
+        cursedText += `${owoify.default(arg, level).replace(/`/g, "\\`")} `;
```
Comment (duncte123) [inline commands/Owoify.js:39]:
```suggestion
        cursedText += `${owoify.default(arg, level)} `;
```

File: commands/Owoify.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        charCount += 1;
+      }
+    });
+
+    if (charCount > 1000) {
+      return message.channel.send("Sorry, but the character limit is 1000~!");
+    }
+
+    return message.channel.send(
+      `OwO-ified for ${message.author}~!\n\n${cursedText}`
```
Comment (duncte123) [inline commands/Owoify.js:52]:
```suggestion
      `OwO-ified for ${message.author}~!\n\n${cursedText.replace(/`/g, "\\`")}`
```

File: commands/Owoify.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        cursedText += `${arg} `;
+        charCount += 1;
+      }
+    });
+
+    if (charCount > 1000) {
+      return message.channel.send("Sorry, but the character limit is 1000~!");
+    }
+
+    return message.channel.send(
```
Comment (duncte123) [inline commands/Owoify.js:51]: Why are you returning here?

File: commands/Owoify.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        cursedText += `${owoify.default(arg, level).replace(/`/g, "\\`")} `;
+        charCount += arg.length;
+      } else {
+        cursedText += `${arg} `;
+        charCount += 1;
+      }
+    });
+
+    if (charCount > 1000) {
+      return message.channel.send("Sorry, but the character limit is 1000~!");
```
Comment (duncte123) [inline commands/Owoify.js:48]: Please don't do this as the command handler will await the promise for sending the message (and node is single-threaded)

File: commands/Owoify.js [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+  constructor() {
+    super({
+      name: "owoify",
+      category: "fun",
+      description: "This command will owoify your text",
+      usage: `Run \`owoify <text>\` or \`owoify <intensity> <text>\` to owoify any text!\nIntensities available are \`easy\` (default), \`medium\`, and \`hard\`.`
+    });
+  }
+
+  async run(client, message, args) {
```
Comment (duncte123) [inline commands/Owoify.js:21]:
```suggestion
  run(client, message, args) {
```

File: commands/Owoify.js [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    let level = "owo";
+
+    if (!args.length) {
+      return message.channel.send(
```
Comment (duncte123) [inline commands/Owoify.js:25]:
```suggestion
      message.channel.send(
```

File: commands/Owoify.js [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+        `${message.author}, can you please provide me a message? <:YoichiPlease:692008252690530334>`
+      );
```
Comment (duncte123) [inline commands/Owoify.js:27]:
```suggestion
      );
      return;
```

File: commands/Owoify.js [hunk 14]
```diff
@@ -0,0 +1,55 @@
+import owoify from "owoify-js";
+import Command from "./base/Command.js";
+import { EMOTE_ID_REGEX } from "../utils/discordPatterns.js";
+
+export default class Owoify extends Command {
+  intensities = {
```
Comment (duncte123) [inline commands/Owoify.js:6]: This should be a private prop
```suggestion
  #intensities = {
```

File: commands/Owoify.js [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    let level = "owo";
+
+    if (!args.length) {
+      return message.channel.send(
+        `${message.author}, can you please provide me a message? <:YoichiPlease:692008252690530334>`
+      );
+    }
+
+    if (this.intensities[args[0]]) {
+      level = this.intensities[args.shift()];
```
Comment (duncte123) [inline commands/Owoify.js:31]:
```suggestion
    if (this.#intensities[args[0]]) {
      level = this.#intensities[args.shift()];
```

File: commands/Owoify.js [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+      level = this.#intensities[args.shift()];
+    }
+
+    let cursedText = "";
+    let charCount = args.map(a => a.length).reduce((a, b) => a + b, 0);
+
+    args.forEach(arg => {
+      if (!EMOTE_ID_REGEX.test(arg)) {
+        cursedText += `${owoify.default(arg, level)} `;
+        charCount += arg.length;
```
Comment (duncte123) [inline commands/Owoify.js:41]:
```suggestion

```

File: commands/Owoify.js [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+      } else {
+        cursedText += `${arg} `;
+        charCount += 1;
```
Comment (duncte123) [inline commands/Owoify.js:44]:
```suggestion

```

File: commands/Owoify.js [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+        cursedText += `${owoify.default(arg, level)} `;
+      } else {
+        cursedText += `${arg} `;
+      }
+    });
+
+    if (charCount > 1000) {
+      message.channel.send("Sorry, but the character limit is 1000~!");
+      return;
+    }
```
Comment (Leguu) [inline commands/Owoify.js:49]: Can we move this check above where we convert the message? No use converting the message only to realize that the character limit is reached.

File: commands/Owoify.js [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+      );
+      return;
+    }
+
+    if (this.#intensities[args[0]]) {
+      level = this.#intensities[args.shift()];
+    }
+
+    let cursedText = "";
+    let charCount = args.map(a => a.length).reduce((a, b) => a + b, 0);
```
Comment (Leguu) [inline commands/Owoify.js:36]: Change this to `const`

**Pull request merged**